packages/apps/desktop-viewer-test/src/frontend/hooks/useDownload.tsx

```diff
@@ -0,0 +1,57 @@
+/*---------------------------------------------------------------------------------------------
+ * Copyright (c) Bentley Systems, Incorporated. All rights reserved.
+ * See LICENSE.md in the project root for license terms and full copyright notice.
+ *--------------------------------------------------------------------------------------------*/
+
+import type { ProgressInfo } from "@bentley/itwin-client";
+import { IModelVersion, SyncMode } from "@itwin/core-common";
+import { NativeApp } from "@itwin/core-frontend";
+import { useCallback, useContext, useState } from "react";
+
+import { ITwinViewerApp } from "../app/ITwinViewerApp";
+import { SettingsContext } from "../services/SettingsClient";
+
+export const useDownload = (
+  iModelId: string,
+  iModelName: string,
+  iTwinId: string
+) => {
+  const [progress, setProgress] = useState<number>();
+  const userSettings = useContext(SettingsContext);
+
+  const addRecent = useCallback(async (fileName: string) => {
+    await userSettings.addRecent(fileName, iModelName, iTwinId, iModelId);
+  }, []);
+
+  const doDownload = useCallback(async () => {
+    const fileName = await ITwinViewerApp.saveBriefcase(iModelName);
+    if (fileName) {
+      const req = await NativeApp.requestDownloadBriefcase(
+        iTwinId as string,
```

Comment:
cast should be uneccessary here, its already of type string

Reply:
fixed